09-intermediate-rails/oauth.md

```diff
@@ -165,6 +165,15 @@ Now everything is in place to initialize a User using the hash that is returned
 class SessionsController < ApplicationController
   def create
     auth_hash = request.env['omniauth.auth']
+
+    if auth_hash['uid']
+      @user = User.find_by(uid: auth_hash[:uid], provider: 'github')
+      if @user.nil?
+      end
+    else
+      flash[:error] = ""
+      redirect_to root_path
+    end
     redirect_to login_failure_path unless auth_hash['uid']
 
     @user = User.find_by(uid: auth_hash[:uid], provider: 'github')
```

Comment:
Looks like you've got this `User.find_by` line in here twice. I suspect this whole method may need some cleanup.